### PR TITLE
Add mocking to provide for real, contained testing

### PR DESCRIPTION
**Pull request by jeffbyrnes**
@eherot I think this is needed before we can actually publish this to the Supermarket. Needs more testing, but I thought I’d ask for your :eyes: on this for the next step(s).

### CODE REVIEW

File: libraries/ec2_dns_server.rb [hunk 1]
```diff
@@ -114,19 +114,43 @@ def connection
...
+      end
+
+      connection
     end
   end
 
+  def mock_servers
+    Fog.mock!
+
+    connection = Fog::Compute.new(
```
Comment (eherot) [inline libraries/ec2_dns_server.rb]: Two problems here:
1. `connection` is a method name in this class and should not be re-used.
2. Most of the code in this method duplicates code in `connection` and IMO, we should just read the "mock" option in `connection` and set AWS API flags accordingly.

Comment (eherot) [inline libraries/ec2_dns_server.rb]: Oh, also, _if_ you were going to keep this design, you'd want to just `return mock_servers`.

Comment (eherot) [inline libraries/ec2_dns_server.rb]: We should actually pass the AWS keys in through parameters, and then do something like this:
```ruby
# set @connection_options in initialize

def connection
  @connection ||= begin
    require 'fog'
    opts = { provider: 'AWS' }.merge(@connection_options)
    if @node['ec2dnsserver']['mocking']
      Fog.mock!
    elsif !opts[:aws_access_key_id]
      opts[use_iam_profile: true]
    end
    Fog::Compute.new opts
  end
end
```

Comment (eherot) [inline libraries/ec2_dns_server.rb]: Alternatively, consider calling `create_mock_env!` from outside the class and use the regular memoized `connection` method in that method.

Comment (jeffbyrnes) [inline libraries/ec2_dns_server.rb]: Thanks much, all great points. I knew I was going about this a bit awkwardly.

Comment (eherot) [inline libraries/ec2_dns_server.rb]: Bump....

Comment (jeffbyrnes) [inline libraries/ec2_dns_server.rb]: heh, yeah.

**Comment by jeffbyrnes** [comment]
This is ridiculously stale. Closing.

**Pull request closed without merging**